src/main/scala/fm/xml/XmlReader.scala

```diff
@@ -64,143 +101,186 @@ object XmlReader {
-   * to something like "www.aftermarket.org" when it should be "http://www.aftermarket.org"
-   */
+    * Similar to DefaultNamespaceStreamReaderDelegate but just completely overrides the namespace.  Used when people set the namespace
+    * to something like "www.aftermarket.org" when it should be "http://www.aftermarket.org"
+    */
   private class OverrideDefaultNamespaceStreamReaderDelegate(self: XMLStreamReader2, overrideDefaultNamespaceURI: String) extends StreamReader2Delegate(self) {
     Predef.require(overrideDefaultNamespaceURI.isNotBlank, "Expected overrideDefaultNamespaceURI to be not blank")
-    
+
     override def getNamespaceURI(): String = {
       // If the prefix is blank (which I thinks means we are using the default namespace) use the override
       if (getPrefix().isBlank) overrideDefaultNamespaceURI else super.getNamespaceURI()
     }
   }
 }
 
+
+object XmlReaderPath {
+  // Apply helper for reading a non-mapped value
+  def apply[XmlValue: ClassTag](itemPath: String): XmlReaderPath[XmlValue, XmlValue] = XmlReaderPath[XmlValue, XmlValue](itemPath, { v: XmlValue => v })
+}
+
 /**
- * defaultNamespaceURI - If no namespace is set, use this as a default
- * overrideDefaultNamespaceURI - Override the namespace and just use this instead
- */
-class XmlReader[T: ClassTag](rootName: String, itemPath: String, defaultNamespaceURI: String, overrideDefaultNamespaceURI: String, protected val resource: Resource[Reader]) extends ResourceLazySeq[T, Reader] with Logging {
-  
-  if (defaultNamespaceURI.isNotBlank) require(overrideDefaultNamespaceURI.isBlank, "Can't set both defaultNamespaceURI and overrideDefaultNamespaceURI")
-  if (overrideDefaultNamespaceURI.isNotBlank) require(defaultNamespaceURI.isBlank, "Can't set both defaultNamespaceURI and overrideDefaultNamespaceURI")
-  
-  private[this] val itemClass: Class[T] = classTag[T].runtimeClass.asInstanceOf[Class[T]]
+  *
+  * @param itemPath The XPath-like path to the element we are interested in.  (e.g. "part", "items/part", etc)
+  * @param toMappedValue Map the XmlValue into another return type, this should match the return type in MultiXmlReader
+  * @tparam XmlValue The unmarshalled element class type
+  * @tparam MappedXmlValue A mapped XmlValue type
+  */
+final case class XmlReaderPath[
+XmlValue: ClassTag,
+MappedXmlValue: ClassTag,
+](itemPath: String, toMappedValue: XmlValue => MappedXmlValue) extends Logging {
+  private[this] val itemClass: Class[XmlValue] = classTag[XmlValue].runtimeClass.asInstanceOf[Class[XmlValue]]
   private[this] val jaxbContext: JAXBContext = JAXBContext.newInstance(itemClass)
-  
+  private[this] val unmarshaller: Unmarshaller = jaxbContext.createUnmarshaller()
+
   // The XPath-like path to the element we are interested in
   // part => Array("part"), items/part => Array("items","part")
-  private[this] val path: Array[String] = itemPath.split('/')
-  
+  private[xml] def path: Array[String] = itemPath.split('/')
```

Comment:
Why is this a `def`?

Reply:
not sure why these got changed, I'll revert back to val